webapp-client/src/component/AddCategoryForm.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+export default class AddCategoryForm extends React.Component {
+  /** This function will be changed later to directly POST request to db instead of calling addCategory */
+  createCategory = (e) => {
+    e.preventDefault();
+    const category = this._inputElement.value;
+    this.props.addCategory(category);
+    this._inputElement.value = '';
+  };
+  render() {
+    return (
+      <form className="my-1" onSubmit={this.createCategory}>
+        <input
+          ref={(input) => (this._inputElement = input)}
```

Comment:
why not just give it a string name for the ref?

Reply:
I found a way to not use ref at all (by getting the input using e.target.elements['add-category'].value) and I currently have the no ref version in the latest commit, but lmk if it is better to use ref.